### PR TITLE
support parallel load with compressed file

### PR DESCRIPTION
**Pull request by jianwan0214**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #8124 #8270

## What this PR does / why we need it:

### CODE REVIEW

**Comment by m-schen** [comment]
you are cheating.
you put the real load work at pre-insert operator. and load operator just send the file position?

why not build a pipeline that:

n * external -> merge -> dispatch to -> n * insert

**Comment by nnsgmsone** [comment]
The correct pipeline should be [external meta-scan -> dispatch-> multiple external scans]

**Review by iamlinjunhong — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

File: pkg/sql/compile/compile.go [hunk 1]
```diff
@@ -645,6 +646,20 @@ func (c *Compile) ConstructScope() *Scope {
 	return ds
 }
 
+func (c *Compile) ConstructMergeScope() *Scope {
+	ds := &Scope{Magic: Merge}
+	ds.Proc = process.NewWithAnalyze(c.proc, c.ctx, 1, c.anal.Nodes())
+	ds.Proc.LoadTag = true
+	ds.Proc.ParallelLoad = true
```
Comment (m-schen) [inline pkg/sql/compile/compile.go]: remove the load related codes outside of ConstructMergeScope() or remove whole method.

**Review by m-schen — APPROVED** [review]

**Review by nnsgmsone — CHANGES_REQUESTED** [review]
算子间交互的永远都是batch，然后externfill为什么不是一个普通的projection

**Comment by nnsgmsone** [comment]
一个更加合理的方式可能是构建plan的时候就得到一个类似这样的plan, extern_scan(得到csv数据) -> projection(用特定函数从csv数据中提取数据） -> insert。然后设置projection和insert的并行度即可

**Review by ouyuanning — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by m-schen — CHANGES_REQUESTED** [review]

**Review by m-schen — APPROVED** [review]
lgtm.

**Review by heni02 — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

File: pkg/sql/colexec/external/external.go [hunk 2]
```diff
@@ -522,6 +524,7 @@ func GetBatchData(param *ExternalParam, plh *ParseLineHandler, proc *process.Pro
 	for k := 0; k < n; k++ {
 		bat.Zs[k] = 1
 	}
+	bat.Cnt = 1
```
Comment (nnsgmsone) [inline pkg/sql/colexec/external/external.go:527]: what is this?

**Review by nnsgmsone — CHANGES_REQUESTED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Comment by guguducken** [comment]
@mergify refresh

**Pull request merged**